libp2p/crypto/crypto.nim

```diff
@@ -786,27 +786,29 @@ proc decodeExchange*(message: seq[byte],
      pb.getLengthValue(2, signature) != -1:
     result = true
 
+{.pop.}
```

Comment:
rather than `{.pop.}` it's better to individually annotate the proc's with what they raise exactly (so it's visible in the signature)

Reply:
Makes sense, other then that are you fine with this solution for now? rather then just `get`?

Reply:
yeah sure - I mean as long as the error is documented in the signature, that's a significant step forwards - in this particular part we're interfacing with an exceptions-based library so it makes sense to raise exceptions for its benefit - `Result` is designed to allow this bridging to happen smoothly